sandy/core/cov.py

```diff
@@ -2075,28 +2124,114 @@ def sparse_tables_cholesky(a, rows=1000):
     os.remove('cholesky.h5')
     return low_triang_matrix
 
-def triu_matrix(arr, size):
+
+def segmented_pivot_table(data_stack, rows = 10000000,
+                          index=["MAT", "MT", "E"],

```

Comment:
I guess here index and columns should not be keyword arguments, but positional arguments, without default